clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -207,9 +207,10 @@ public synchronized void ensureCoordinatorReady() {
     }
 
     protected RequestFuture<Void> lookupCoordinator() {
-        if (findCoordinatorFuture == null) {
-            findCoordinatorFuture = sendGroupCoordinatorRequest();
-            findCoordinatorFuture.addListener(new RequestFutureListener<Void>() {
+        RequestFuture<Void> future = findCoordinatorFuture;
+        if (future == null) {
+            findCoordinatorFuture = future = sendGroupCoordinatorRequest();
```

Comment:
@hachikuji In the previous PR, you thought it was worthwhile to null out the Future after usage (i.e. after              `client.poll(future)` returns). Shall we still do that?